lib/std/os.zig

```diff
@@ -2706,6 +2706,58 @@ pub fn socket(domain: u32, socket_type: u32, protocol: u32) SocketError!socket_t
     }
 }
 
+pub const ShutdownError = error{
+    ConnectionAborted,
+
+    /// Connection was reset by peer, application should close socket as it is no longer usable.
+    ConnectionResetByPeer,
+
+    BlockingOperationInProgress,
+
+    /// The network subsystem has failed.
+    NetworkSubsystemFailed,
+
+    /// The socket is not connected (connection-oriented sockets only).
+    SocketNotConnected,
+
+    SystemResources
+} || UnexpectedError;
+
+pub const ShutdownHow = enum {
```

Comment:
I'd make this a simple Zig enum `enum { read_end, write_end, both_ends }` and convert this to the proper flag value in `fn shutdown`.

Reply:
I dunno, this seems simpler than adding conversion logic.

Reply:
It's a leaky abstraction. The "conversion" code is a simple switch, the same you're doing here with a lot of `if`s.

Reply:
ok I've changed it to switches internal to the function.  This seems like it could be adding unnecessary overhead to me, but not a big deal.